shell/components/CruResource.vue

```diff
@@ -257,10 +266,32 @@ export default {
       this.$emit('select-type', id);
     },
 
+    clickSave(e) {
+      this.createNamespaceIfNeeded();
```

Comment:
This is emitting 'finish' even if `createNamespaceIfNeeded` throws an error, so the non-namespace resource save function still runs and returns a 'not found' error because the namespace its referencing doesn't exist. That error is what's shown in the form, which is misleading. 
I think `createNamespaceIfNeeded` needs to return a promise so we can catch the error and emit 'error' instead of 'finish'